streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -290,6 +290,10 @@ public void buildAndOptimizeTopology(final Properties props) {
             }
 
             if (streamGraphNode.allParentsWrittenToTopology() && !streamGraphNode.hasWrittenToTopology()) {
+                if (props != null && !props.isEmpty()) {
+                    internalTopologyBuilder.setStreamsConfig(new StreamsConfig(props));
```

Comment:
I think we should unify this can call this method only once.

Why not just call it with `KafkaStreams` constructor?

Reply:
Ksql calls the `StreamsBuilder` first, which calls the `StreamStreamJoinNode.writeToTopology()` at some point. When `KafkaStreams` is called later, the shared state store was already added to the join nodes.